### PR TITLE
Add minLSTM model

### PR DESCRIPTION
**Pull request by mbrotos**
https://github.com/lucidrains/minGRU-pytorch/blob/70b258b7e497efcb27997172e40e01d05b5acce9/minGRU_pytorch/minGRULM.py#L63

The minLSTM model can be hot-swapped for minGRU in the language modelling module.

### CODE REVIEW

**Comment by lucidrains** [comment]
@mbrotos ohhh nice! have you tried running this?

**Comment by mbrotos** [comment]
> @mbrotos ohhh nice! have you tried running this?

Yes, I ran about ~200 steps on `enwik8` and the training loss and outputs look comparable to minGRU.

The only meaningful difference is when I compute `log_h0`:
https://github.com/lucidrains/minGRU-pytorch/blob/a5ca670d06fb39d09fb168a1a9d17fa36e082765/minGRU_pytorch/minLSTM.py#L48

vs. in minGRU: 
https://github.com/lucidrains/minGRU-pytorch/blob/70b258b7e497efcb27997172e40e01d05b5acce9/minGRU_pytorch/minGRU.py#L63

Is there a reason why you only take the `log` of the previous hidden state rather `log_g` activation defined in the paper?:
image

--

Lmk if you'd like me to expand the PR, otherwise I feel it could be useful for others who want the implementation.

**Comment by lucidrains** [comment]
@mbrotos it has been a while, but i found an [old self-explanatory comment](https://github.com/lucidrains/minGRU-pytorch/issues/9#issuecomment-2411533990)

**Comment by lucidrains** [comment]
@mbrotos thank you so much, graciously accept this 🙏
